cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -820,6 +844,30 @@ protected Future<Boolean> delete(Reconciliation reconciliation) {
                 .map(Boolean.FALSE); // Return FALSE since other resources are still deleted by garbage collection
     }
 
+    /**
+     * Remove the metrics specific to the kind implementing it.
+     *
+     * @param desiredNames  Set of resources which should be reconciled
+     * @param namespace     The namespace to reconcile, or {@code *} to reconcile across all namespaces.
+     */
+    @Override
+    public void removeMetrics(Set<NamespaceAndName> desiredNames, String namespace) {
+        // Removing all CA cluster metrics not in the desired set
+        Set<String> desiredClusterNames = desiredNames.stream().map(NamespaceAndName::getName).collect(Collectors.toSet());
+        metrics.removeMetricsForCertificates(key -> {
+            if (!key.getKind().equals(kind())) {
+                return false;
+            }
+            if (namespace.equals(ConfigParameter.ANY_NAMESPACE)) {
+                // If all namespaces, remove everything for that Kind
+                return true;
+            }
+
+            // Otherwise only remove the ones related to the namespace and not in the desired set
+            return key.getNamespace().equals(namespace) && !desiredClusterNames.contains(key.getClusterName());
```

Comment:
@scholzj I rewrote logic to cater specifically for `*` in `namespace`, do you agree with this now, that its indeed what we discussed?

Reply:
@scholzj @maciej-tatarski  got some time tonight to test it and it looks like i works in multi cluster scenario now as well 😄 All seems to be good.